src/hotspot/cpu/x86/x86_32.ad

```diff
@@ -13150,7 +13150,27 @@ instruct cmovII_mem_LTGE(cmpOp cmp, flagsReg_long_LTGE flags, rRegI dst, memory
   ins_pipe( pipe_cmov_mem );
 %}
 
-// Compare 2 longs and CMOVE ints.
+instruct cmovII_reg_LTGE_U(cmpOpU cmp, flagsReg_ulong_LTGE flags, rRegI dst, rRegI src) %{
+  predicate(VM_Version::supports_cmov() && ( _kids[0]->_kids[0]->_leaf->as_Bool()->_test._test == BoolTest::lt || _kids[0]->_kids[0]->_leaf->as_Bool()->_test._test == BoolTest::ge ));
+  match(Set dst (CMoveI (Binary cmp flags) (Binary dst src)));
+  ins_cost(200);
+  format %{ "CMOV$cmp $dst,$src" %}
+  opcode(0x0F,0x40);
+  ins_encode( enc_cmov(cmp), RegReg( dst, src ) );
+  ins_pipe( pipe_cmov_reg );
```

Comment:
Since this code is shared with `cmovII_reg_LTGE`, we could directly use it with `expand`:
```
  expand %{
    cmovII_reg_LTGE(cmp, flags, dst, src);
  %}
```
Could also be done for the other cases.

Reply:
Thanks Christian, that's a good suggestion. I thought expand does not work because the argument types differ but it does. I updated the patch.

Reply:
Looks good! AFAICT the two operands `flagsReg_(u)long_LTGE` only differ in their format string which is probably why it works. But maybe someone else can jump in here who knows the code better.